index.js

```diff
@@ -159,22 +160,81 @@ module.exports = {
       })
     }
 
-    function publish(content, cb) {
-      if (cb === undefined) return promisify(publish)(content)
+    function excludeMembers(groupId, feedIds, opts = {}, cb) {
+      if (cb === undefined)
+        return promisify(excludeMembers)(groupId, feedIds, opts)
+
+      const excludeContent = {
+        type: 'group/exclude',
+        excludes: feedIds,
+        recps: [groupId],
+      }
+      const excludeOpts = { tangles: ['group', 'members'], spec: () => true }
+      publish(excludeContent, excludeOpts, (err, exclusionMsg) => {
+        // prettier-ignore
+        if (err) return cb(clarify(err, 'Failed to publish exclude msg'))
+
+        pull(
+          listMembers(groupId),
+          pull.collect((err, beforeMembers) => {
+            if (err) return cb(err)
+
+            const remainingMembers = beforeMembers.filter(
+              (member) => !feedIds.includes(member)
+            )
+            const newGroupKey = new SecretKey()
+            const addInfo = { key: newGroupKey.toBuffer() }
+
+            ssb.box2.addGroupInfo(groupId, addInfo, (err) => {
+              if (err) return cb(err)
+
+              const newKey = {
+                key: newGroupKey.toBuffer(),
+                scheme: keySchemes.private_group,
+              }
+              ssb.box2.pickGroupWriteKey(groupId, newKey, (err) => {
+                if (err) return cb(err)
+
+                const newKeyContent = {
+                  type: 'group/move-epoch',
+                  secret: newGroupKey.toString('base64'),
+                  exclusion: fromMessageSigil(exclusionMsg.key),
+                  recps: [groupId, ...remainingMembers],
+                }
+                // TODO: loop if many members
+                // TODO: post this on old feed
+                publish(newKeyContent, { spec: () => true }, (err) => {
+                  // prettier-ignore
+                  if (err) return cb(clarify(err, 'Failed to tell people about new epoch'))
+
+                  return cb()
+                })
+              })
+            })
+          })
+        )
+      })
+    }
+
+    function publish(content, opts, cb) {
+      if (cb === undefined) return promisify(publish)(content, opts)
 
       if (!content) return cb(new Error('Missing content'))
 
+      const isSpec = opts?.spec ?? isContent
+      const tangles = opts?.tangles ?? ['group']
```

Comment:
Maybe it's better to only let people append stuff to this array, otherwise they might accidentally remove the `group` tangle. idk if there's any situation where that tangle shouldn't be there in a group situation

Reply:
done in d8baa13